google-cloud-errors/lib/google/cloud/errors.rb

```diff
@@ -119,19 +121,26 @@ def status_details
       end
 
       ##
-      # Returns the `::Google::Rpc::ErrorInfo` object present in `status_details` array,
-      # given that the following is true:
-      #   * `status_details` exists and is an array
-      #   * there is exactly one `::Google::Rpc::ErrorInfo` object in the `status_details` array
+      # Returns the `::Google::Rpc::ErrorInfo` object present in the `status_details`
+      # or `details` array, given that the following is true:
+      #   * either `status_details` or `details` exists and is an array
+      #   * there is exactly one `::Google::Rpc::ErrorInfo` object in that array.
+      # Looks in `status_details` first, then in `details`.
       #
       # @return [::Google::Rpc::ErrorInfo, nil]
       def error_info
-        @error_info ||= if status_details.is_a? Array
-                          error_infos = status_details.find_all { |status| status.is_a?(::Google::Rpc::ErrorInfo) }
-                          if error_infos.length == 1
-                            error_infos[0]
-                          end
-                        end
+        @error_info ||= begin
+          check_property = lambda do |prop|
+            if prop.is_a? Array
+              error_infos = prop.find_all { |status| status.is_a?(::Google::Rpc::ErrorInfo) }
+              if error_infos.length == 1
+                error_infos[0]
+              end
+            end
+          end
+
+          check_property.call(status_details) || check_property.call(details)
```

Comment:
Is there a reason to create a lambda here? Why not just write a private method?

Reply:
private method would only be used here in one line so lambda seemed like a natural fit